assets/ts/controllers/component/checkbox_select_all_controller.ts

```diff
@@ -2,18 +2,71 @@ import {Controller} from '@hotwired/stimulus';
 
 export default class extends Controller {
     public static targets = ['toggle'];
-    private readonly declare toggleTargets: HTMLInputElement[];
 
     public toggleAll(event: Event): void {
-        const target = event.currentTarget as HTMLInputElement;
-        const role   = target.dataset.role;
+        const target    = event.currentTarget as HTMLInputElement;
+        const role      = target.dataset.forRole;
+        const search    = this.element.querySelector<HTMLInputElement>('[data-role~="search"]')?.value.toLowerCase() ?? '';
+        const revisions = this.getRevisions();
 
-        if (role === undefined) {
-            this.toggleTargets.forEach(el => el.checked = target.checked);
+        if (search === '' || search === undefined) {
+            revisions.forEach(el => el.querySelector<HTMLInputElement>(`[data-role~="${role}"]`)!.checked = target.checked);
         } else {
-            this.element
-                .querySelectorAll<HTMLInputElement>(`[data-role~="${role}"]`)
-                .forEach(el => el.checked = target.checked);
+            revisions.forEach(el => el.querySelector<HTMLInputElement>(`[data-role~="${role}"]`)!.checked = this.matchesSearch(el.closest('[data-role~="revision"]'), search) && target.checked);
         }
     }
+
+    public search(event: Event): void {
```

Comment:
Can this be separated to separate controller? You can add multiple stimulus controllers to the same element.

Reply:
To combine the "toggle all" functionality with the visible revisions. use the `checkVisibility` browser function to check if the element is visible.